### PR TITLE
RHCLOUD-31123: Add support for pgsql16 in local db provider

### PR DESCRIPTION
**Pull request by bsquizz**

### CODE REVIEW

**Review by adamrdrew — APPROVED** [review]
LGTM

File: controllers/cloud.redhat.com/providers/utils/utils.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ var DefaultImageDatabasePG12 = "quay.io/cloudservices/postgresql-rds:12-2318dee"
 var DefaultImageDatabasePG13 = "quay.io/cloudservices/postgresql-rds:13-2318dee"
 var DefaultImageDatabasePG14 = "quay.io/cloudservices/postgresql-rds:14-2318dee"
 var DefaultImageDatabasePG15 = "quay.io/cloudservices/postgresql-rds:15-2318dee"
+var DefaultImageDatabasePG16 = "quay.io/cloudservices/postgresql-rds:16-f145533"
```
Comment (psav) [inline controllers/cloud.redhat.com/providers/utils/utils.go:36]: Should the others be updated too??

Comment (bsquizz) [reply controllers/cloud.redhat.com/providers/utils/utils.go:36]: No need, main change was adding pgsql16 support. I am hoping to not trigger a DB update for all the others.

**Pull request merged**